fontc_crater/src/ci.rs

```diff
@@ -246,3 +263,44 @@ fn format_elapsed_time<Tmz: TimeZone>(start: &DateTime<Tmz>, end: &DateTime<Tmz>
     write!(&mut out, "{secs}s").unwrap();
     out
 }
+
+fn precompile_rust_binaries(temp_dir: &Path) -> (PathBuf, PathBuf) {
+    fn copy_file_into_dir(file_path: &Path, dir_path: &Path) -> PathBuf {
+        let new_file_path = dir_path.join(file_path.file_name().unwrap());
+        std::fs::copy(file_path, &new_file_path).expect("failed to copy binary to tempdir");
+        new_file_path
+    }
```

Comment:
I find this style *far* harder to read than top-level functions, can we push this up a level to stand alone?